src/main/java/org/openrewrite/github/UseGradleWrapper.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.github;
+
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.*;
+import org.openrewrite.yaml.JsonPathMatcher;
+import org.openrewrite.yaml.YamlIsoVisitor;
+import org.openrewrite.yaml.tree.Yaml;
+
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+// We don't use the atomicity of AtomicBoolean, just the mutability
+public class UseGradleWrapper extends ScanningRecipe<AtomicBoolean> {
+    @Override
+    public String getDisplayName() {
+        return "Use Gradle Wrapper instead of Gradle binary directly";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replace calls to `gradle` with calls to `gradlew` (Gradle Wrapper) in any `.github/workflows/*.yml` file.";
+    }
+
+    @Override
+    public AtomicBoolean getInitialValue(ExecutionContext ctx) {
+        return new AtomicBoolean(false);
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getScanner(AtomicBoolean acc) {
+        return new TreeVisitor<Tree, ExecutionContext>() {
+            @Override
+            public Tree visit(Tree tree, ExecutionContext ctx, Cursor parent) {
+                acc.set(Paths.get("gradlew").toFile().exists());
```

Comment:
Not sure if this is the proper way. Can we assume OpenRewrite is called in the project root directory?
Or do we have some "constant" to refer to here?

Reply:
Perhaps it helps to look at how other recipes identify the Gradle wrapper, [like UpdateGradleWrapper](https://github.com/openrewrite/rewrite/blob/main/rewrite-gradle/src/main/java/org/openrewrite/gradle/UpdateGradleWrapper.java).

Reply:
I agree, this should be looking at the source path of the incoming source files to see if the components of a Gradle wrapper are present. It can then use those confirmations later during the edit phase. 

You should _NOT_ rely on local disk operations.